sui-execution/latest/sui-move-natives/src/light_client/tendermint_lc.rs

```diff
@@ -46,9 +56,49 @@ pub fn tendermint_state_proof(
     ty_args: Vec<Type>,
     mut args: VecDeque<Value>,
 ) -> PartialVMResult<NativeResult> {
-    todo!()
+    let value = pop_arg!(args, Vector).to_vec_u8()?;
```

Comment:
can we use "destructing" pattern? See:
* https://doc.rust-lang.org/rust-by-example/flow_control/match/destructuring/destructure_slice.html
* https://stackoverflow.com/a/72552078